runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/block/Block.java

```diff
@@ -120,12 +118,13 @@
    * If another element is written after this method is called, a new non-committed partition should be created
    * for the element even if a partition with the same key is committed already.
    *
-   * @throws BlockWriteException for any error occurred while trying to commit partitions.
+   * @throws org.apache.nemo.common.exception.BlockWriteException for any error occurred
```

Comment:
For the methods that have no `throws` anymore, could you remove the lines for `@throws` in the comments?
Yet we still need the explanation so maybe we could add something like this:

For 
```
/**
   * Commits this block to prevent further write.
   *
   * @return the size of each partition if the data in the block is serialized.
   * @throws BlockWriteException for any error occurred while trying to commit a block.
   *                             (This exception will be thrown to the scheduler
   *                             through {@link org.apache.nemo.runtime.executor.Executor} and
   *                             have to be handled by the scheduler with fault tolerance mechanism.)
   */
```

may be changed to:
```
/**
   * Commits this block to prevent further write.
   * Classes that implement this interface may throw {@link BlockWriteException}
   * for any error occurred while trying to commit a block.
   * (This exception will be thrown to the scheduler through {@link org.apache.nemo.runtime.executor.Executor}
   * and have to be handled by the scheduler with fault tolerance mechanism.)
   *
   * @return the size of each partition if the data in the block is serialized.
   */
```

Reply:
Also, instead of writing `org.apache.nemo.common.exception.BlockWriteException`, you may use {@link ...} :)

Reply:
Resolved